lib/trino-filesystem-manager/src/main/java/io/trino/filesystem/manager/FileSystemModule.java

```diff
@@ -162,6 +165,13 @@ static TrinoFileSystemFactory createFileSystemFactory(
 
         TrinoFileSystemFactory delegate = new SwitchingFileSystemFactory(loader);
         delegate = new TracingFileSystemFactory(tracer, delegate);
+
+        // Enable leak detection if configured or if running in a CI environment
+        boolean trackingDetectionEnabled = getenv("CONTINUOUS_INTEGRATION") != null;
```

Comment:
You can make this the default for the config, allowing it to be overridden in either case.

Reply:
Will follow up